examples/mugen/retrieval/video_clip.py

```diff
@@ -68,18 +75,37 @@ def forward(self, input_ids: torch.Tensor) -> torch.Tensor:
         return last_hidden_state[:, self.target_token_idx, :]
 
 
-class VideoEncoder(nn.Module):
-    """Encode videos to a fixed size vector. Adapted from VideoCLIP
-        (https://github.com/facebookresearch/fairseq/blob/main/examples/MMPT/mmpt/processors/models/s3dg.py)
+class VideoEncoder(nn.Module, PretrainedMixin):
+    """Encode videos to a fixed size vector.
+
+    Adapted from MUGEN's video encoder
+        (https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/models/videoclip/modules.py)
+
+    Args:
+        pretrained (bool): whether to use a pretrained model or not.
+            Defaults to True.
+        pretrain_path (str): local path or remote URL to pretrained weights.
+            Defaults to ``PRETRAINED_S3D_KINETICS400_URL``, the weights MUGEN used from
+            pretraining S3D on Kinetics 400. Ignored if ``pretrained`` is False.
 
     Inputs:
         x (Tensor): batch of videos with dimensions (batch, channel, time, height, width)
     """
 
-    def __init__(self):
+    def __init__(
+        self,
+        pretrained: bool = True,
```

Comment:
Non blocking Q:

logic related to loading checkpoints --- should they be part of the model code or elevated to the builder? See #130 . Let's start to think about this as a design question. You will be able to revise this in your builder PR (if needed).